triple-lift.js

```diff
@@ -375,51 +540,44 @@ window.headertag.partnerScopes.push(function() {
              * format:
              *
              *     {
-             *         <htSlotId>: {
-             *             demand: {
-             *                 <key>: <value>,
-             *                 <key>: <value>,
-             *                 ...
-             *             }
-             *         },
-             *         ...
+             *         slot: {
+             *             <htSlotId>: {
+             *                 timestamp: Utils.now(),
+             *                 demand: {
+             *                     <key>: <value>,
+             *                     <key>: <value>,
+             *                     ...
+             *                 }
+             *             },
+             *             ...
+             *         }
              *     }
              */
 
             /* PUT CODE HERE */
 
-            /* -------------------------------------------------------------------------- */
-
-        }
+            // map slots to TripleLift xSlots and make request for each placement
+            var numPlacements = Object.keys(slots).length;
+            slots.forEach(function(slot) {
+                var slotId = slot.getSlotElementId();
+                mapping[slotId].forEach(function(xSlotId) {
```

Comment:
This function will throw an error if a particular slot doesn't have a corresponding mapping for it. Should add a check for whether the slotId exists in the mapping.